tests/test_cms_menus.py

```diff
@@ -250,6 +253,115 @@ def test_get_roots_with_versioning_disabled(self):
+        )
+        pagecontent_bbb = factories.PageContentWithVersionFactory(
+            language=self.language,
+            version__created_by=self.get_superuser(),
+            title="bbb",
+            menu_title="bbb",
+            page_title="bbb",
+            version__state=PUBLISHED,
+        )
+
+        child = factories.ChildMenuItemFactory(parent=menu_cont_draft.root, content=pagecontent_aaa.page)
+        child_1 = factories.ChildMenuItemFactory(parent=menu_cont_published.root, content=pagecontent_bbb.page)
+
+        # Node added in draft menu version is rendered in page draft view only
+        draft_page_endpoint = get_object_edit_url(pagecontent_aaa)
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(draft_page_endpoint)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(child.title, str(response.content))
+        self.assertNotIn(child_1.title, str(response.content))
+
+    def test_draft_menu_on_published_page(self):
+        """
+         Draft menu changes on published page
+        """
+        menu_cont_draft = factories.MenuContentWithVersionFactory(version__state=DRAFT, language=self.language)
+        menu_cont_published = factories.MenuContentWithVersionFactory(version__state=PUBLISHED,
+                                                                      language=self.language)
+        pagecontent_aaa = factories.PageContentWithVersionFactory(
+            language=self.language,
+            version__created_by=self.get_superuser(),
+            title="aaa",
+            menu_title="aaa",
+            page_title="aaa",
+            version__state=DRAFT,
+        )
+        pagecontent_bbb = factories.PageContentWithVersionFactory(
+            language=self.language,
+            version__created_by=self.get_superuser(),
+            title="bbb",
+            menu_title="bbb",
+            page_title="bbb",
+            version__state=PUBLISHED,
+        )
+
+        child = factories.ChildMenuItemFactory(parent=menu_cont_draft.root, content=pagecontent_aaa.page)
+        child_1 = factories.ChildMenuItemFactory(parent=menu_cont_published.root, content=pagecontent_bbb.page)
+
+        # Node added in draft menu version is not rendered in published view, only published menu nodes are rendered
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.get(pagecontent_bbb.page.get_absolute_url())
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(child_1.title, str(response.content))
+        self.assertNotIn(child.title, str(response.content))
+
+    def test_draft_menu_in_preview_mode(self):
+        """
+        rendering draft poge in preview mode will show published menu
+        """
+        menu_cont_published = factories.MenuContentWithVersionFactory(version__state=PUBLISHED, language=self.language)
+        pagecontent_aaa = factories.PageContentWithVersionFactory(
```

Comment:
This test will only work if the pages share the same grouper (page). The published version has to be added first too as this is logically how the versions are created.